src/draw.c

```diff
@@ -764,7 +764,7 @@ static void render_content(cairo_t *c, struct colored_layout *cl, int width, dou
                 cairo_set_line_width(c, frame_width * scale);
                 draw_rounded_rect(c,
                                 frame_x + half_frame_width,
-                                frame_y + half_frame_width,
```

Comment:
With cairo you have to specify the center of the line to draw. So you want to place the line half of the width down from where you want the top to be. Otherwise you get blurry lines (see picture where line width is 1). Maybe the fill needs to be moved down.

![Screenshot_20230619_213904](https://github.com/dunst-project/dunst/assets/23078054/8c54f101-54b4-4a35-8a72-0679cc9dd3db)